physical/raft/raft_network_test.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+//go:build !race
+
+// The raft networking layer tends to reset the TLS keyring, which triggers
+// the race detector even though it should be a no-op.
+
+package raft
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul/sdk/freeport"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/sdk/physical"
+	"github.com/hashicorp/vault/vault/cluster"
+)
+
+func BenchmarkRaftWithNetwork(b *testing.B) {
+	b.StopTimer()
+	raft1, _ := createRaftNetworkCluster(b, true, false)
+	b.StartTimer()
+	for i := 0; i < b.N; i++ {
+		err := raft1.Put(context.Background(), &physical.Entry{
+			Key:   "test",
+			Value: []byte{byte(i)},
+		})
+		if err != nil {
+			b.Fatal(err)
+		}
+		snapOut, err := os.CreateTemp(b.TempDir(), "bench")
+		if err != nil {
+			b.Fatal(err)
+		}
+		b.Cleanup(func() {
+			_ = snapOut.Close()
+			_ = os.Remove(snapOut.Name())
+		})
+		err = raft1.Snapshot(snapOut, nil)
+		if err != nil {
+			b.Fatal(err)
+		}
+	}
+}
+
+// TestRaftNetworkClusterWithMultipleTimeEncodingsSet tests that Raft nodes
+// with different msgpack time.Time encodings set will still cluster together.
+// However, with go-msgpack 2.1.0+, the decoder is tolerant of both encodings,
+// so this could only fail if the decoder drastically changes in the future.
+func TestRaftNetworkClusterWithMultipleTimeEncodingsSet(t *testing.T) {
```

Comment:
It seems like this test is verifying that we don't accidentally make a change that results in Vault using the old timestamp format.  But isn't the only way that could happen if we tweak the setting that exists solely for the sake of allowing this test to work?  I'm not sure that it's worth keeping this test on that basis.

Reply:
:+1: